src/lib/users/ssh_authorized_keyring.rb

```diff
@@ -115,27 +116,39 @@ def empty?
         keys.empty?
       end
 
+      # Determines if the keyring has changed
+      #
+      # @return [Boolean] +true+ if it has changed; +false+ otherwise
+      def changed?
+        @keys != @old_keys
+      end
+
       # Read keys from a given home directory and add them to the keyring
       #
       # @return [Array<String>] List of authorized keys
       def read_keys
         path = authorized_keys_path
-        @keys = FileUtils::Exists(path) ? SSHAuthorizedKeysFile.new(path).keys : []
-        log.info "Read #{@keys.size} keys from #{path}"
-        @keys
+        @old_keys = FileUtils::Exists(path) ? SSHAuthorizedKeysFile.new(path).keys : []
+        log.info "Read #{@old_keys.size} keys from #{path}"
+        @keys = @old_keys.dup
       end
 
       # Write user keys to the given file
       #
       # If SSH_DIR does not exist in the given directory, it will be
       # created inheriting owner/group and setting permissions to SSH_DIR_PERM.
       def write_keys
+        return unless changed?
+
         remove_authorized_keys_file
+
```

Comment:
np: why this blank line?

Reply:
Nothing special. I just think that it helps to read the code better. I'll keep it, hope you don't mind :)